gtsfm/view_graph_estimator/cycle_consistent_rotation_estimator.py

```diff
@@ -0,0 +1,231 @@
+        max_gt_error_in_cycle: List[float],
+        per_edge_aggregate_error: Dict[Tuple[int, int], float],
+        two_view_reports_dict: Dict[Tuple[int, int], TwoViewEstimationReport],
+    ) -> None:
+        """Saves plots of aggregate error vs GT error for each edge, and cyclic error vs max GT error for each cycle.
+
+        Args:
+            inlier_edges: Set of all cycle consistent edges.
+            cycle_errors: Cyclic error for all cycles.
+            max_gt_error_in_cycle: Maximum GT rotation error in the cycle, for all cycles.
+            per_edge_aggregate_error: Dict from edge index pair to aggregate cyclic error of the edge.
+            two_view_reports_dict: Dict from edge index pair to the TwoViewEstimationReport of the edge.
+        """
+        # aggregate info over per edge_errors
+        inlier_errors_aggregate = []
+        inlier_errors_wrt_gt = []
+
+        outlier_errors_aggregate = []
+        outlier_errors_wrt_gt = []
+        for (i1, i2), error_aggregate in per_edge_aggregate_error.items():
+            if (i1, i2) in inlier_edges:
+                inlier_errors_aggregate.append(error_aggregate)
+                inlier_errors_wrt_gt.append(two_view_reports_dict[(i1, i2)].R_error_deg)
+            else:
+                outlier_errors_aggregate.append(error_aggregate)
+                outlier_errors_wrt_gt.append(two_view_reports_dict[(i1, i2)].R_error_deg)
+        plt.scatter(
+            inlier_errors_aggregate,
+            inlier_errors_wrt_gt,
+            10,
+            color="g",
+            marker=".",
+            label=f"inliers @ {MAX_INLIER_MEASUREMENT_ERROR_DEG} deg.",
+        )
+        plt.scatter(
+            outlier_errors_aggregate,
+            outlier_errors_wrt_gt,
+            10,
+            color="r",
+            marker=".",
+            label=f"outliers @ {MAX_INLIER_MEASUREMENT_ERROR_DEG} deg.",
+        )
+        plt.xlabel(f"{self._edge_error_aggregation_criterion} cycle error")
+        plt.ylabel("Rotation error w.r.t GT")
+        plt.axis("equal")
+        plt.legend(loc="lower right")
+        plt.savefig(os.path.join("plots", f"gt_err_vs_{self._edge_error_aggregation_criterion}_agg_error.jpg"), dpi=400)
+        plt.close("all")
+
+        plt.scatter(cycle_errors, max_gt_error_in_cycle)
+        plt.xlabel("Cycle error")
+        plt.ylabel("Avg. Rot3 error over cycle triplet")
+        plt.axis("equal")
+        plt.savefig(os.path.join("plots", "cycle_error_vs_GT_rot_error.jpg"), dpi=400)
+        plt.close("all")
+
+    def __get_valid_input_edges(self, i2Ri1: Dict[Tuple[int, int], Rot3]) -> List[Tuple[int, int]]:
+        """Gets the input edges (i1, i2) with the relative rotation i2Ri1 where:
+        1. i1 < i2
+        2. i2Ri1 is not None
+
+        Args:
+            i2Ri1: input dictionary of relative rotations.
+
+        Returns:
+            List of valid edges.
+        """
+        valid_edges = []
+        for (i1, i2), i2Ri1 in i2Ri1.items():
```

Comment:
@akshay-krishnan we need to rename this variable otherwise we are overwriting it in the loop
```python
for (i1, i2), i2Ri1 in i2Ri1.items():
```